samples/sensor/bq274xx/src/main.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+#include <sys/printk.h>
+
+static void do_main(struct device *dev)
+{
+	struct sensor_value voltage, current, state_of_charge,
+		 full_charge_capacity, remaining_charge_capacity, avg_power,
+		 int_temp, current_standby, current_max_load,
+		 state_of_health;
+
+	int status = 0;
+
+	while (1) {
+		status = sensor_sample_fetch(dev);
+		if (status < 0) {
+			printk("Unable to fetch the samples\n");
+		}
+
+		status = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_VOLTAGE,
+				&voltage);
+		if (status < 0) {
+			printk("Unable to get the voltage value\n");
+		}
+
+		printk("Voltage: %d.%06dV\n", voltage.val1,
+				voltage.val2);
+
+		status = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_AVG_CURRENT,
+				&current);
+		if (status < 0) {
+			printk("Unable to get the current value\n");
+		}
+
+		if ((current.val2 < 0) && (current.val1 >= 0)) {
+			current.val2 = -(current.val2);
+			printk("Avg Current: -%d.%06dA\n", current.val1,
+					current.val2);
+		} else if ((current.val2 > 0) && (current.val1 < 0)) {
+			printk("Avg Current: %d.%06dA\n", current.val1,
+					current.val2);
+		} else if ((current.val2 < 0) && (current.val1 < 0)) {
+			current.val2 = -(current.val2);
+			printk("Avg Current: %d.%06dA\n", current.val1,
+					current.val2);
+		} else {
+			printk("Avg Current: %d.%06dA\n", current.val1,
+					current.val2);
+		}
+
+		status = sensor_channel_get(dev, SENSOR_CHAN_GAUGE_STDBY_CURRENT,
+				&current_standby);
+		if (status < 0) {
+			printk("Unable to get the current value\n");
+		}
+
+		if ((current_standby.val2 < 0) && (current_standby.val1 >= 0)) {
+			current_standby.val2 = -(current_standby.val2);
+			printk("Standby Current: -%d.%06dA\n", current_standby.val1,
+					current_standby.val2);
+		} else if ((current_standby.val2 > 0) && (current_standby.val1 < 0)) {
+			printk("Standby Current: %d.%06dA\n", current_standby.val1,
+					current_standby.val2);
+		} else if ((current_standby.val2 < 0) && (current_standby.val1 < 0)) {
+			current_standby.val2 = -(current_standby.val2);
+			printk("Standby Current: %d.%06dA\n", current_standby.val1,
+					current_standby.val2);
+		} else {
+			printk("Standby Current: %d.%06dA\n", current_standby.val1,
+					current_standby.val2);
+		}
```

Comment:
extract this into a function so you don't have to repeat it multiple times

Reply:
Abstracted out. Thanks